autorest/azure/environments.go

```diff
@@ -126,5 +135,17 @@ func EnvironmentFromName(name string) (Environment, error) {
 	if !ok {
 		return env, fmt.Errorf("autorest/azure: There is no cloud environment matching the name %q", name)
 	}
+
+	if strings.EqualFold(name, "AZURESTACKCLOUD") == true {
+		fbytes, err := ioutil.ReadFile(path.Join("/etc", strings.ToLower(name), "azurestackcloud.json"))
```

Comment:
Ah, I'd rather see this getting passed in as parameter, with the `"/etc/<blah>/azurestackcloud.json"`  used as a default in some capacity.